mecanum_drive_controller/include/mecanum_drive_controller/odometry.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2022, Stogl Robotics Consulting UG (haftungsbeschränkt)
+// (template)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MECANUM_DRIVE_CONTROLLER__ODOMETRY_HPP_
+#define MECANUM_DRIVE_CONTROLLER__ODOMETRY_HPP_
+
+// #include <ros/time.h>
+#include "realtime_tools/realtime_buffer.h"
+#include "realtime_tools/realtime_publisher.h"
+
+#include <boost/accumulators/accumulators.hpp>
+#include <boost/accumulators/statistics/rolling_mean.hpp>
+#include <boost/accumulators/statistics/stats.hpp>
+#include <boost/function.hpp>
+
+#define PLANAR_POINT_DIM 3
+
+namespace mecanum_drive_controller
+{
+namespace bacc = boost::accumulators;
+
+/// \brief The Odometry class handles odometry readings
+/// (2D pose and velocity with related timestamp)
+class Odometry
+{
+public:
+  /// Integration function, used to integrate the odometry:
+  typedef boost::function<void(double, double, double)> IntegrationFunction;
```

Comment:
It should be working also like this

```suggestion
  typedef std::function<void(double, double, double)> IntegrationFunction;
```